sdk/src/workflow_context.rs

```diff
@@ -556,16 +556,16 @@ impl<'a> Future for LATimerBackoffFut<'a> {
         }
         let poll_res = self.current_fut.poll_unpin(cx);
         if let Poll::Ready(ref r) = poll_res {
-            // If we've already said we want to cancel, don't schedule the backoff timer. Just
-            // return cancel status. This can happen if cancel comes after the LA says it wants to
-            // back off but before we have scheduled the timer.
-            if self.did_cancel.load(Ordering::Acquire) {
-                return Poll::Ready(ActivityResolution {
-                    status: Some(activity_resolution::Status::Cancelled(Default::default())),
-                });
-            }
-
             if let Some(activity_resolution::Status::Backoff(b)) = r.status.as_ref() {
+                // If we've already said we want to cancel, don't schedule the backoff timer. Just
+                // return cancel status. This can happen if cancel comes after the LA says it wants
+                // to back off but before we have scheduled the timer.
+                if self.did_cancel.load(Ordering::Acquire) {
+                    return Poll::Ready(ActivityResolution {
+                        status: Some(activity_resolution::Status::Cancelled(Default::default())),
+                    });
+                }
```

Comment:
Any consequences to moving this under the conditional? Seems before we'd return cancel if cancel was requested whereas now we only will if cancel was requested _and_ it's backing off.

Reply:
This is in the Rust SDK, but yeah it was actually wrong before. This was only supposed to deal with early return in specifically the backoff case, and it returning a Default value was hiding real cancel reasons.